mm2src/coins/eth/web3_transport.rs

```diff
@@ -159,13 +174,39 @@ async fn send_request(
     const REQUEST_TIMEOUT_S: f64 = 60.;
 
     let mut errors = Vec::new();
-    for uri in uris.iter() {
-        let request = to_string(&request);
-        event_handlers.on_outgoing_request(request.as_bytes());
 
-        let mut req = http::Request::new(request.clone().into_bytes());
+    let serialized_request = to_string(&request);
+
+    for node in nodes.iter() {
+        let mut serialized_request = serialized_request.clone();
+        if node.gui_auth {
+            if let Some(generator) = gui_auth_validation_generator.clone() {
+                let mut json_payload: Json = serde_json::from_str(&serialized_request)?;
```

Comment:
Serializing the payload back and forth is overhead. Is it possible to use `#[serde(flatten)]` instead? E.g.
```rust
struct AuthPayload {
    #[serde(flatten)]
    request: Call,
    signed_message: SignedMessage
}
```

Reply:
Yes, that works. Very nice workaround :) Thanks!